### PR TITLE
Add init helm charts

### PR DESCRIPTION
**Pull request by ishankhare07**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement
/kind feature

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
Adds support to initialize a newly created vcluster with helm charts

**Please provide a short message that should be published in the vcluster release notes**
Add capability where vcluster and be initialized with helm charts.

**What else do we need to know?**

### CODE REVIEW

File: pkg/controllers/manifests/init.go [hunk 1]
```diff
@@ -96,9 +148,397 @@ func (r *InitManifestsConfigMapReconciler) Reconcile(ctx context.Context, req ct
...
+	lastAppliedConfig := cm.ObjectMeta.Annotations[LastAppliedChartConfig]
+	// first check if its the chart version that has changed
+	// as that would mean we first need to trigger a new archive to be pulled
+	lastConfigRaw, err := compress.Uncompress(lastAppliedConfig)
+	if err != nil {
+		r.Log.Errorf("unable to uncompress last applied config: %v", err)
+	}
+
+	lastConfig := make(map[string]string)
+	err = json.Unmarshal([]byte(lastConfigRaw), &lastConfig)
```
Comment (FabianKramm) [inline pkg/controllers/manifests/init.go:268]: I guess its better to use bundle hash here only as the bundle is already compressed and would make up quite a big portion of the annotation here

File: pkg/controllers/manifests/init.go [hunk 2]
```diff
@@ -96,9 +148,397 @@ func (r *InitManifestsConfigMapReconciler) Reconcile(ctx context.Context, req ct
...
+	tarball, err := base64.StdEncoding.DecodeString(bundle)
+	if err != nil {
+		r.Log.Errorf("unable to decode bundle: %v", err)
+		return "", err
+	}
+
+	chartName := cm.Data["name"]
+	// version, _ := cm.Data["version"]
+
+	path, err := tarutil.ExtractTarGz(chartName, bytes.NewReader(tarball))
```
Comment (FabianKramm) [inline pkg/controllers/manifests/init.go:494]: why extracting, you can also use `helm install test my-tar.tgz`

**Review by FabianKramm — CHANGES_REQUESTED** [review]
@ishankhare07 thanks for the PR! Please see my comments below. Would be also good if you add an e2e test that verifies this is correctly working.

File: Dockerfile [hunk 3]
```diff
@@ -8,6 +8,9 @@ ARG TARGETARCH
 # Install kubectl for development
 RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl && chmod +x ./kubectl && mv ./kubectl /usr/local/bin/kubectl
 
+# Install helm binary
+RUN curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 && chmod 700 get_helm.sh && ./get_helm.sh
```
[the remark this answers is not in the document]
Comment (ishankhare07) [reply Dockerfile:12]: I did test that and seems to work fine with all the e2e tests passing.

File: pkg/controllers/manifests/init.go [hunk 4]
```diff
@@ -96,9 +145,414 @@ func (r *InitManifestsConfigMapReconciler) Reconcile(ctx context.Context, req ct
...
+		r.Log.Errorf("unable to get compressed config for chart %s: %v", name, err)
+		return err
+	}
+
+	if cm.ObjectMeta.Annotations == nil {
+		cm.ObjectMeta.Annotations = map[string]string{}
+	}
+
+	cm.ObjectMeta.Annotations[LastAppliedChartConfig] = compressedConfig
+	err = r.LocalClient.Update(ctx, cm, &client.UpdateOptions{})
```
Comment (FabianKramm) [inline pkg/controllers/manifests/init.go:336]: You don't have to pass the updateoptions here

File: pkg/controllers/manifests/init.go [hunk 5]
```diff
@@ -96,9 +145,414 @@ func (r *InitManifestsConfigMapReconciler) Reconcile(ctx context.Context, req ct
...
+	// during the next reconcile
+	tarball := fmt.Sprintf("%s-%s.tgz", name, version)
+	rawTarBall, err := os.ReadFile(tarball)
+	if err != nil {
+		r.Log.Errorf("error reading in pulled archive: %v", err)
+		return err
+	}
+
+	cm.Data["bundle"] = base64.StdEncoding.EncodeToString(rawTarBall)
+	err = r.LocalClient.Update(ctx, cm, &client.UpdateOptions{})
```
Comment (FabianKramm) [inline pkg/controllers/manifests/init.go:462]: Remove update options here

**Review by FabianKramm — CHANGES_REQUESTED** [review]
@ishankhare07 a couple of smaller things, please take a look at my comments below

File: devspace.yaml [hunk 6]
```diff
@@ -49,6 +49,17 @@ deployments:
             create: true
           role:
             extended: true
+        init:
```
[the remark this answers is not in the document]
Comment (ishankhare07) [reply devspace.yaml:52]: moving this to the `deploy` profile patches

**Review by FabianKramm — APPROVED** [review]
@ishankhare07 thanks for the PR! LGTM!

**Pull request merged**